WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingView.swift

```diff
@@ -5,53 +5,69 @@ struct StoreOnboardingView: View {
-            // Progress view
-            StoreSetupProgressView(isExpanded: viewModel.isExpanded,
-                                   totalNumberOfTasks: viewModel.taskViewModels.count,
-                                   numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
-
-            // Task list
-            VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
-                ForEach(Array(viewModel.taskViewModels.enumerated()), id: \.offset) { index, taskViewModel in
-                    let isLastTask = index == viewModel.taskViewModels.count - 1
-
-                    StoreOnboardingTaskView(viewModel: taskViewModel,
-                                            showDivider: !isLastTask) { task in
-                        taskTapped(task)
+        VStack {
+            Color(.systemColor(.systemGray6))
+                .frame(height: 16)
+                .renderedIf(!viewModel.isExpanded)
+
+            let verticalSpacing = viewModel.isExpanded ? Layout.VerticalSpacing.expandedMode : Layout.VerticalSpacing.collapsedMode
+            VStack(alignment: viewModel.isExpanded ? .center : .leading, spacing: verticalSpacing) {
+                DismissButton(action: dismissAction)
+                    .renderedIf(viewModel.isExpanded)
+
+                // Progress view
+                StoreSetupProgressView(isExpanded: viewModel.isExpanded,
+                                       totalNumberOfTasks: viewModel.taskViewModels.count,
+                                       numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
+
+                // Task list
+                VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
+                    ForEach(Array(viewModel.tasksForDisplay.enumerated()), id: \.offset) { index, taskViewModel in
+                        let isLastTask = index == viewModel.tasksForDisplay.count - 1
+
+                        StoreOnboardingTaskView(viewModel: taskViewModel,
+                                                showDivider: !isLastTask) { task in
+                            taskTapped(task)
+                        }
                     }
                 }
-            }
 
-            // View all button
-            Button {
-                viewAllTapped?()
-            } label: {
-                Text(String(format: Localization.viewAll, viewModel.taskViewModels.count))
-                    .fontWeight(.semibold)
-                    .foregroundColor(.init(uiColor: .accent))
-                    .subheadlineStyle()
+                // View all button
+                ViewAllButton(action: viewAllTapped, text: String(format: Localization.viewAll, viewModel.taskViewModels.count))
+                    .renderedIf(!viewModel.isExpanded)
+
+                Spacer()
+                    .renderedIf(viewModel.isExpanded)
             }
-            .renderedIf(!viewModel.isExpanded)
+            .padding(insets: Layout.insets)
```

Comment:
💡 I would also suggest adding a background color for this `VStack` so that the content looks nicer in dark mode:

```suggestion
            .padding(insets: Layout.insets)
            .background(Color(uiColor: .listForeground(modal: false)))
```

<img src="https://user-images.githubusercontent.com/5533851/221763838-1f95d0db-5d32-4e0b-b9f3-13f405b763ba.png" width=320 />

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8991/commits/00ea28f2c5b58b022f2efb8b52edca067b7d7425